class/practice_2.2/main.cpp

```diff
@@ -0,0 +1,126 @@
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <vector>
+#include <string>
+
+using namespace std;
+
+struct Pixel {
+    int r, g, b;
+
+    bool operator==(const Pixel& other) const {
+        return r == other.r && g == other.g && b == other.b;
+    }
+};
+
+const int WIDTH = 5;
+const int HEIGHT = 5;
+
+bool readImage(const string& filename, vector<vector<Pixel>>& image, const string& outputFilename) {
+    ifstream inFile(filename);
+    ofstream outFile(outputFilename);
+    if (!inFile) {
+        cerr << "Unable to open input file." << endl;
+        return false;
+    }
+
+    string line;
+    int row = 0;
+    while (getline(inFile, line)) {
+        stringstream ss(line);
+        vector<Pixel> pixelRow;
+        for (int col = 0; col < WIDTH; ++col) {
+            Pixel pixel;
+            char comma;
+            ss >> pixel.r >> comma >> pixel.g >> comma >> pixel.b;
+            if (comma != ',' || pixel.r < 0 || pixel.r > 255 || pixel.g < 0 || pixel.g > 255 || pixel.b < 0 || pixel.b > 255) {
+                cerr << "Invalid input format at row " << row + 1 << ", column " << col + 1 << "." << endl;
+                return false;
+            }
+            pixelRow.push_back(pixel);
+        }
+        if (pixelRow.size() != WIDTH) {
+            cerr << "Input file should contain exactly " << WIDTH << " pixels per line." << endl;
+            return false;
+        }
+        image.push_back(pixelRow);
+        ++row;
+    }
+
+    if (row != HEIGHT || image.size() != HEIGHT) {
+        cerr << "Input file should contain exactly " << HEIGHT << " lines." << endl;
+        if (outFile) {
+            outFile << "Input file should contain exactly " << HEIGHT << " lines.";
+        }
+        return false;
+    }
+
+    return true;
+}
+
+
+void writeImage(const string& filename, const vector<vector<Pixel>>& image) {
+    ofstream outFile(filename);
+    if (!outFile) {
+        cerr << "Unable to open output file." << endl;
+        return;
+    }
+
+    for (const auto& row : image) {
+        for (size_t col = 0; col < row.size(); ++col) {
+            outFile << row[col].r << ',' << row[col].g << ',' << row[col].b;
```

Comment:
std::format and join for array should do the trick if you will extend Pixel class accordingly. Not sure whether it is supported fully in the current version of compiler but you should try https://en.cppreference.com/w/cpp/ranges/join_with_view